test/e2e/scheduling/preemption.go

```diff
@@ -152,13 +159,20 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 		// Create one pod per node that uses a lot of the node's resources.
 		By("Create pods that use 60% of node resources.")
 		pods := make([]*corev1.Pod, len(nodeList.Items))
+		allPods, err := cs.CoreV1().Pods(metav1.NamespaceAll).List(metav1.ListOptions{})
 		for i, node := range nodeList.Items {
-			cpuAllocatable, found := node.Status.Allocatable["cpu"]
+			currentCpuUsage, currentMemUsage := getCurrentPodUsageOnTheNode(node.Name, allPods.Items, podRequestedResource)
+			framework.Logf("Current cpu usage and memory usage is %v, %v", currentCpuUsage, currentMemUsage)
+			currentNode, err := cs.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
```

Comment:
This actually is the crux of the change, previously, we were using node.allocatable but the node object is the same in every iteration which is causing the tests to fail on heavily loaded nodes.

@bsalamat

Reply:
I am not sure if I understand what the problem is. Are you saying that other tests running at the same time (in parallel) to our tests cause the flakes?
That aside, when nodes have the same amount of allocatable (which they often do), this PR will be the same as the existing version of the test, unless I am missing something.